h5py/_hl/dataset.py

```diff
@@ -35,10 +35,11 @@
 def make_new_dset(parent, shape=None, dtype=None, data=None, name=None,
                   chunks=None, compression=None, shuffle=None,
                   fletcher32=None, maxshape=None, compression_opts=None,
-                  fillvalue=None, scaleoffset=None, track_times=False,
-                  external=None, track_order=None, dcpl=None, dapl=None,
-                  efile_prefix=None, virtual_prefix=None, allow_unknown_filter=False,
-                  rdcc_nslots=None, rdcc_nbytes=None, rdcc_w0=None):
+                  fillvalue=None, fill_time=None, scaleoffset=None,
```

Comment:
This change is API breaking: we cannot add new arguments at arbitrary positions between two existing positionally-allowed arguments.

Please move this new argument to the end of the signature and make it keyword-only.

Reply:
Thanks for pointing this out, I haven't considered this.